### PR TITLE
feat: control log level verbosity

### PR DESCRIPTION
**Pull request by dkontorovskyy**
- [ ] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [X] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [X] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does
Control log level verbosity

### CODE REVIEW

File: snyk-monitor/templates/deployment.yaml [hunk 1]
```diff
@@ -58,6 +58,8 @@ spec:
             value: {{ .Values.https_proxy }}
           - name: NO_PROXY
             value: {{ .Values.no_proxy }}
+          - name: LOG_LEVEL
+            value: {{ .Values.log_level }}
```
Comment (agranado2k) [inline snyk-monitor/templates/deployment.yaml:62]: I guess you need to include the same options to the helm templates [here](https://github.com/snyk/kubernetes-monitor/blob/ecca7f5ce26fc8926660871ed07896a3109199be/snyk-monitor/templates/deployment.yaml#L59) for `snyk-monitor/templates/deployment.yaml`

**Review by agranado2k — COMMENTED** [review]
I have a comment! :)
Also I believe we should have at least one test to make sure it's working in the way we expect!

File: snyk-monitor/templates/deployment.yaml (hunk 1 above)
Comment (dkontorovskyy) [reply snyk-monitor/templates/deployment.yaml:62]: Your link is pointing to the same place it's in PR, am I missing smth?

File: README.md [hunk 2]
```diff
@@ -140,6 +140,14 @@ no_proxy: *.example.local,*.other.global,192.168.0.0/16
 no_proxy: long.domain.name.local,example.local
 ```
 
+## Changing default log level ##
+
+To lower `snyk-monitor`'s logging verbosity `log_level` value could be set to one of these options:
+* `'WARN'`
+* `'ERROR'`
+
+By default, `log_level` is `'INFO'`
```
Comment (shaimendel) [inline README.md:149]: I'm really sorry about this - but can you please add a `.` in the end? :) to match the style of the docs

File: README.md [hunk 3]
```diff
@@ -140,6 +140,14 @@ no_proxy: *.example.local,*.other.global,192.168.0.0/16
 no_proxy: long.domain.name.local,example.local
 ```
 
+## Changing default log level ##
```
Comment (shaimendel) [inline README.md:143]: maybe remove the `default` part?

Comment (shaimendel) [inline README.md:143]: need to add this also to https://github.com/snyk/kubernetes-monitor/blob/staging/snyk-monitor/README.md with concrete helm example (this file is for YAML installation)

File: src/common/logger.ts [hunk 4]
```diff
@@ -1,6 +1,17 @@
 import * as bunyan from 'bunyan';
 import config = require('./config');
 
+if (process.env.LOG_LEVEL) {
+  try {
+    // Validate if LOG_LEVEL has valid bunyan logging level
+    bunyan.resolveLevel(process.env.LOG_LEVEL);
+  } catch {
+    delete process.env.LOG_LEVEL;
```
Comment (shaimendel) [inline src/common/logger.ts:9]: I know it is a problematic context but can we log something here? so the customer knows what happened

File: src/common/logger.ts [hunk 5]
```diff
@@ -1,6 +1,17 @@
...
+  } finally {
+    config.LOGGING.level = process.env.LOG_LEVEL || config.LOGGING.level;
```
Comment (shaimendel) [inline src/common/logger.ts:11]: can't this be moved to after the `bunyan.resolveLevel` call in the try clause? so we don't need finally at all.
```js
try {
    // Validate if LOG_LEVEL has valid bunyan logging level
    bunyan.resolveLevel(process.env.LOG_LEVEL);
    config.LOGGING.level = process.env.LOG_LEVEL;
  } catch {
    delete process.env.LOG_LEVEL;
  }
```

also - not sure why we need/want to delete the env var

File: src/common/logger.ts [hunk 6]
```diff
@@ -1,6 +1,13 @@
 import * as bunyan from 'bunyan';
 import config = require('./config');
 
+try {
+  // Validate if LOG_LEVEL has valid bunyan logging level
+  config.LOGGING.level = bunyan.resolveLevel(process.env.LOG_LEVEL || config.LOGGING.level);
+} catch(e) {
+  console.error(`Log level "${process.env.LOG_LEVEL}" is not valid logging level. Falling back to "INFO"`);
```
Comment (shaimendel) [inline src/common/logger.ts:8]: That's good, just need to take into account that it won't match the "schema" of the rest of the logs, but I don't think we have a choice here

**Review by shaimendel — APPROVED** [review]

**Pull request merged**